app/models/comment.rb

```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :user, class_name: 'User', foreign_key: 'author_id'
+  belongs_to :post, class_name: 'Post', foreign_key: 'post_id'
+
+  def update_comment_counter
+    post.update(comments_counter: post.comments.count)
+  end
```

Comment:
- [ Optional ] Kindly I recommend you  use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_comments_counter method every time a new comment has been saved to database, then make the update_comments_counter method private Just like this :- 

  # app/models/comment.rb
```
  after_save :update_comments_counter

  private

  def update_comments_counter
       post.update(comments_counter: post.comments.count)
  end
```